### ISSUE TITLE
Task description in Key distribution Kata is not correct

### ISSUE THREAD

**Issue by crazy4pi314** [microsoft/QuantumKatas#295]
I found that tasks 2.4 and 2.5 in the [Key Distribution Kata](https://github.com/microsoft/QuantumKatas/tree/master/KeyDistribution_BB84) are a bit confusing, and I wanted to make a few suggestions as a researcher in QKD. 

Firstly, task 2.5 is _not_ verifying that the key is secure. Task 2.5 really is just checking the error rate/key yield of the raw key exchange process. The error rate is one of multiple parameters and assumptions that go into verifying if the key is secure.

Secondly, it seems to make more logical sense given 2.5 calculates an error rate to swap the order of 2.4 and 2.5 since 2.4 currently yield the distilled key and would come at the very end of any classical post processing steps.

**Comment by tcNickolas** [comment]
I agree, task 2.5 could use a better task title.

It might be the case that task 2.4 needs a better title and details as well, but I'm not sure they should be swapped. Right now 2.4 filters out the measurement results where measurement basis used by A and B don't match; in the end-to-end scenario this has to happen before task 2.5 in which A and B use subsets of the results of filtering to calculate error rate.

(Sorry for the delay replying! Last week was crazier than usual, and I haven't looked at this kata for a while... I really hope to catch up with the issues and pull requests this week!)

**Comment by crazy4pi314** [comment]
I'll take a stab at a new task title and leave the order alone for now.

**Comment by tcNickolas** [comment]
Closing per #297.
